WinUIGallery/App.xaml.cs

```diff
@@ -1,288 +1,273 @@
+            //}
+
+            if (System.Diagnostics.Debugger.IsAttached)
+            {
+                this.DebugSettings.BindingFailed += DebugSettings_BindingFailed;
+            }
+#endif
+
+            EnsureWindow();
+        }
+
+        private void DebugSettings_BindingFailed(object sender, BindingFailedEventArgs e)
+        {
+
+        }
+
+#if WINUI_PRERELEASE
+        protected override void OnActivated(IActivatedEventArgs args)
+        {
+            EnsureWindow(args);
+        }
+#endif
+
+        private async void EnsureWindow(IActivatedEventArgs args = null)
+        {
+            // No matter what our destination is, we're going to need control data loaded - let's knock that out now.
+            // We'll never need to do this again.
+            await ControlInfoDataSource.Instance.GetGroupsAsync();
+            await IconsDataSource.Instance.LoadIcons();
+
+            Frame rootFrame = GetRootFrame();
+
+            ThemeHelper.Initialize();
+
+            Type targetPageType = typeof(NewControlsPage);
+            string targetPageArguments = string.Empty;
+
+            if (args != null)
+            {
+                if (args.Kind == ActivationKind.Launch)
+                {
+                    if (args.PreviousExecutionState == ApplicationExecutionState.Terminated)
+                    {
+                        try
+                        {
+                            await SuspensionManager.RestoreAsync();
+                        }
+                        catch (SuspensionManagerException)
+                        {
+                            //Something went wrong restoring state.
+                            //Assume there is no state and continue
+                        }
+                    }
+
+                    targetPageArguments = ((Windows.ApplicationModel.Activation.LaunchActivatedEventArgs)args).Arguments;
+                }
+            }
+            var eventargs = Microsoft.Windows.AppLifecycle.AppInstance.GetCurrent().GetActivatedEventArgs();
+            if (eventargs != null && eventargs.Kind is ExtendedActivationKind.Protocol && eventargs.Data is ProtocolActivatedEventArgs)
+            {
+                ProtocolActivatedEventArgs ProtocolArgs = eventargs.Data as ProtocolActivatedEventArgs;
+                var uri = ProtocolArgs.Uri.LocalPath.Replace("/", "");
+
+                targetPageArguments = uri;
+                string targetId = string.Empty;
+                File.WriteAllText("D:\\uri.txt", uri);
+
+                if (uri == "AllControls")
+                {
+                    targetPageType = typeof(AllControlsPage);
+                }
+                else if (uri == "NewControls")
+                {
+                    targetPageType = typeof(NewControlsPage);
+                }
+                else if (ControlInfoDataSource.Instance.Groups.Any(g => g.UniqueId == uri))
+                {
+                    targetPageType = typeof(SectionPage);

```

Comment:
Nit: `NavigationView` selection doesn't seem to work for e.g. the Colors/Icons pages. Not a super big deal though, so nothing blocking.

![image](https://github.com/microsoft/WinUI-Gallery/assets/9866362/77c9d309-a0a6-41a6-8e2e-fd757912681c)

Reply:
Good catch, we can also fix this in this PR if you prefer. That way this topic should be done for some time then.

Reply:
Sure!